schedule.py

```diff
@@ -182,16 +184,18 @@ def read_input(input_file):
         if not any(course_name in professor.capabilities for professor in professors.values()):
             raise ValueError('No professor can teach ' + course_name)
 
-    # check that the total number of units from professors is enough
-    # TODO this may not be true in the future, if we just through all available courses into the solver
-    # put in all available courses and select ones with professor preference/set if the course must be offered
+    # check if the required units are more than professors' total units
     total_professor_units = sum(professor.max_units for professor in professors.values())
-    total_course_units = sum(section.units for section in sections.values())
-    if total_course_units > total_professor_units:
-        raise ValueError('Professors can only teach {} units but there are {} course units total'.format(
-            total_professor_units,
-            total_course_units,
-        ))
+    course_units_required = sum(section.units for section in sections.values() if section.must_offer)
+    course_units_optional = sum(section.units for section in sections.values() if not section.must_offer)
```

Comment:
Okay, but I think the representation needs to change - I can imagine a semester requiring 2 sections of a course (which we do with CS1 and Data Structures), but can go up to 3 sections. So `must_offer` should be an integer, not a boolean.